akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -117,6 +117,15 @@ object ShardRegion {
     def shardId(message: Any): String
   }
 
+  /**
+   * Interface of function to generate actor `Props`
+   * from entity id.
+   */
+  @FunctionalInterface
+  trait PropsFactory {
```

Comment:
why do we need this instead of using `java.util.function.Function`?

Reply:
@patriknw In the first commit (before @johanandren's comments) I had this interface for Java and a type alias for Scala, to keep in line with the other params (ExtractEntityId, MessageExtractor, etc.). I also think having a specific type for it makes it clearer what it's for. But it isn't necessary, I can change it if you'd like.

Reply:
I don't think it makes it clearer for such a simple function, it's another level of indirection to look at to understand what the parameter is about

Reply:
I agree, I just didn't think that second step away from the original code :)